### PR TITLE
Mapgen: Spread both night and day light banks in spreadLight

### LINKED ISSUE #3930: No light spread at night in lua mapgens
In a dark cave in my 'noisegrid' lua mapgen https://github.com/paramat/noisegrid

screenshot_20160329_154207

^ In daytime

screenshot_20160329_154223

^ Now change time to night

Light sources spread their light in daytime, but change time to night and only the light source itself is lit, there is no light spread.
I guess it wouldn't affect core mapgen lava because that is always flowing and therefore updating lighting. However if core mapgen ever had glow ore these would have this problem, so i would consider this essential.
My mod has paramtype = "light" for these light sources, essential for spreading light in daytime.

### PR DESCRIPTION
**Pull request by kwolekr**
This fixes #3930.

### CODE REVIEW

**Comment by paramat** [comment]
:+1: Tested with noisegrid mod and works.

File: src/mapgen.cpp [hunk 1]
```diff
@@ -242,18 +242,38 @@ void Mapgen::setLighting(u8 light, v3s16 nmin, v3s16 nmax)
...
-	light--;
-	// should probably compare masked, but doesn't seem to make a difference
-	if (light <= nn.param1 || !ndef->get(nn).light_propagates)
+	u8 light_night = light & 0xF0;
+	if (light_night > 0)
+		light_night -= 0x10;
+
+	// Bail out only if we have no more light from either bank to propogate, or
+	// we hit a solid block that light cannot pass through.
+	if ((light_day  <= (n.param1 & 0x0F) &&
```
Comment (kwolekr) [inline src/mapgen.cpp]: Aware, did that so the <= lines up with the one below it.

Comment (paramat) [inline src/mapgen.cpp]: Ok

**Comment by kwolekr** [comment]
https://github.com/minetest/minetest/commit/0338c2eb91ca2ba19551aaa23bbfb46552cee9ac

**Pull request closed without merging**